modules/react/src/deckgl.ts

```diff
@@ -82,22 +82,26 @@ function createDeckInstance<ViewsT extends ViewOrViews>(
     // The Deck's animation loop is independent from React's render cycle, causing potential
     // synchronization issues. We provide this custom render function to make sure that React
     // and Deck update on the same schedule.
-    _customRender: redrawReason => {
-      // Save the dirty flag for later
-      thisRef.redrawReason = redrawReason;
-
-      // Viewport/view state is passed to child components as props.
-      // If they have changed, we need to trigger a React rerender to update children props.
-      const viewports = deck.getViewports();
-      if (thisRef.lastRenderedViewports !== viewports) {
-        // Viewports have changed, update children props first.
-        // This will delay the Deck canvas redraw till after React update (in useLayoutEffect)
-        // so that the canvas does not get rendered before the child components update.
-        thisRef.forceUpdate();
-      } else {
-        redrawDeck(thisRef);
-      }
-    }
+    // TODO(ibgreen) - Hack to enable WebGPU as it needs to render quickly to avoid CanvasContext texture from going stale
+    _customRender:
+      props.deviceProps?.adapters?.[0]?.type === 'webgpu'
```

Comment:
Nit: the indentation is getting really deep, maybe not do this check inline...

Reply:
Yeah, I tried to move it out but it is referencing deck.